pkg/jsonexp/expressions.go

```diff
@@ -0,0 +1,178 @@
+package jsonexp
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/tidwall/gjson"
+)
+
+type Operator int8
+
+const (
+	UnknownOperator Operator = iota
+	EqualOperator
+	NotEqualOperator
+	IncludesOperator
+	ExcludesOperator
+	RegexOperator
+)
+
+func (o *Operator) String() string {
+	switch *o {
+	case EqualOperator:
+		return "eq"
+	case NotEqualOperator:
+		return "neq"
+	case IncludesOperator:
+		return "incl"
+	case ExcludesOperator:
+		return "excl"
+	case RegexOperator:
+		return "matches"
+	}
+	return "unknown"
+}
+
+func OperatorFromString(operator string) Operator {
+	switch operator {
+	case "eq":
+		return EqualOperator
+	case "neq":
+		return NotEqualOperator
+	case "incl":
+		return IncludesOperator
+	case "excl":
+		return ExcludesOperator
+	case "matches":
+		return RegexOperator
+	}
+	return UnknownOperator
+}
+
+type Pattern struct {
+	Selector string
+	Operator Operator
+	Value    string
+}
+
+func (p Pattern) Matches(json string) (bool, error) {
+	expectedValue := p.Value
+	obtainedValue := gjson.Get(json, p.Selector)
+
+	switch p.Operator {
+	case EqualOperator:
+		return (expectedValue == obtainedValue.String()), nil
+
+	case NotEqualOperator:
+		return (expectedValue != obtainedValue.String()), nil
+
+	case IncludesOperator:
+		for _, item := range obtainedValue.Array() {
+			if expectedValue == item.String() {
+				return true, nil
+			}
+		}
+		return false, nil
+
+	case ExcludesOperator:
+		for _, item := range obtainedValue.Array() {
+			if expectedValue == item.String() {
+				return false, nil
+			}
+		}
+		return true, nil
+
+	case RegexOperator:
+		if re, err := regexp.Compile(expectedValue); err != nil {
+			return false, err
+		} else {
```

Comment:
prob can drop the else here?

Reply:
It requires expanding the scope of `re` and `err`, but yes.